### PR TITLE
Keep all pairs in delta delta correlations

### PR DESCRIPTION
**Pull request by londumas**
Fix issue https://github.com/igmhub/picca/issues/509, by keeping all possible pairs between delta and fitting the contribution of sky residuals in fitter2 by a function that is zero everywhere, except where rp==0.

Here are the results of the fit to the three correlations presented in https://github.com/igmhub/picca/issues/509, computed on the (rp,rt) grid assuming MgII absorption:
- MgII11 = left of MgII emission line: [260,276] nm
- calib = right of MgII emission line: [290,312] nm

The fit runs for all pairs from 10 h-1Mpc to 160h-1Mpc.
Currently the function is a power-law with two parameters, but it doesn't seem to do perfectly the trick.
We have to work on a better function.
It flattens out all small separation.

```
 ||                           || 10^{4} (b\beta)_{\mathrm{MgII(2796)}}/f || gamma_sky   || scale_sky   ||                      || 
 || MGII(MGII11)xMGII(MGII11) || 0.0 +/- 21.0         || -1.582 +/- 0.045     || 0.00091 +/- 0.00012  || 1358.53 / (1252-3),  p = 0.016 || 
 || MGII(calib)xMGII(calib)   || 0.0 +/- 16.0         || -1.485 +/- 0.044     || 0.000536 +/- 7.2e-5 || 1394.34 / (1252-3),  p = 0.0024 || 
 || MGII(MGII11)xMGII(calib)  || -0.0 +/- 5.7         || -1.464 +/- 0.023     || 0.000843 +/- 6.1e-5 || 2842.28 / (2504-3),  p = 1.8e-6 || 
```

systematic_fit_rp_0

systematic_fit_rp_1

### CODE REVIEW

File: py/picca/fitter2/xi.py [hunk 1]
```diff
@@ -231,6 +231,30 @@ def qso_bias_vs_z_croom(z, tracer, zref = None, **kwargs):
     p1 = kwargs["croom_par1"]
     return (p0 + p1*(1.+z)**2)/(p0 + p1*(1+zref)**2)
 
+def sky_residual_correlation(r, mu, name, bin_size_rp, *pars, **kwargs):
+    '''Contribution to the correlation of delta field
```
Comment (ngbusca) [inline py/picca/fitter2/xi.py]: @londumas could you instead add an option to the current broadband scheme: e.g. `pre-sky` and call the function `sky_residual...` for this option? Also, could you rename `sky_residual_correlation` to `broadband_sky_res`?

**Comment by ngbusca** [comment]
By the way, the proposed changes do not keep the backwards compatible option to remove the same-plate pairs (once merged into master we won't be able to run the DR14 analysis under the same conditions). Maybe we should simply change the option `no-same-wavelength-pairs` to `--keep-same-wavelength-pairs` so that the default analysis keeps them, but we can always go back to rejection. What do you think @londumas ?

**Comment by ngbusca** [comment]
Also, it would be good to do a test run of this branch on the full data to see whether things improve or at least that they don't change dramatically.

**Comment by londumas** [comment]
@ngbusca, Yes good idea. Let me do that.

**Comment by londumas** [comment]
@ngbusca, Could you review the PR. I applied all your corrections.
And I reproduce the unittest on my machine.

Sur l'auto-correlation des forets Lya:

```
 ||                      || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias-hcd             || BB-cf_z_0_10-0-broadband_sky-scale-sky || BB-cf_z_0_10-0-broadband_sky-sigma-sky ||                      || 
 ||                      ||                      ||                      ||                      ||                      ||                      ||                      ||                      ||                      || 
 || std                  || 24.77 +/- 0.58       || 1.764 +/- 0.080      || 0.68 +/- 0.17        || -0.2287 +/- 0.0040   || -0.0292 +/- 0.0028   ||  +/- 0.0             ||  +/- 0.0             || 1513.16 / (1252-5),  p = 2.8e-07000 || 
 || std+fitRes           || 24.78 +/- 0.60       || 1.484 +/- 0.065      || 0.84 +/- 0.17        || -0.2056 +/- 0.0041   || -0.0257 +/- 0.0024   || 0.000121 +/- 1.1e-050 || 11.50 +/- 0.62       || 1256.54 / (1252-7),  p = 0.40 ||
```

Sur celle des MgII:

```
 ||                      || scale-sky || sigma-sky || 10^4 * (b\beta)_{\mathrm{MgII(2796)}}/f ||                      || 
 ||                      ||                      ||                      ||                      ||                      || 
 || MGII(MGII11)xMGII(MGII11) || 1.99e-5 +/- 1.4e-6 || 7.03 +/- 0.30        || -29.2 +/- 7.2        || 1547.53 / (1590-3),  p = 0.76 || 
 ||                      || scale-sky || sigma-sky || 10^4 * (b\beta)_{\mathrm{MgII(2796)}}/f ||                      || 
 ||                      ||                      ||                      ||                      ||                      || 
 || MGII(calib)xMGII(calib) || 1.28e-5 +/- 8.6e-7 || 8.12 +/- 0.35        || -25.5 +/- 5.8        || 1647.43 / (1590-3),  p = 0.14 || 
 ||                      || scale-sky || sigma-sky || 10^4 * (b\beta)_{\mathrm{MgII(2796)}}/f ||                      || 
 ||                      ||                      ||                      ||                      ||                      || 
 || MGII(MGII11)xMGII(calib) || 2.536e-5 +/- 9e-7 || 7.31 +/- 0.16        || -22.1 +/- 5.2        || 3225.74 / (3180-3),  p = 0.27 ||
```

fit_residuals_mgii

**Comment by ngbusca** [comment]
@londumas thanks, looks great! I'm surprised by the ndf in the fits. Could you re-post them with the standard cuts 10 < r [Mpc/h] < 180 ?

**Comment by ngbusca** [comment]
Also, could you post the rp == 0 bin for the LyA auto data and best fit for:
* fit using the `rp==0` bin, no broadband
* fit excluding `rp ==0` bin, no broadband
* fit including `rp==0` bin, with broadband

**Comment by londumas** [comment]
@ngbusca, here it is:
```
 ||                      || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias-hcd             ||                      || 
 ||                      ||                      ||                      ||                      ||                      ||                      ||                      || 
 || std                  || 24.73 +/- 0.58       || 1.774 +/- 0.079      || 0.67 +/- 0.17        || -0.229 +/- 0.0040    || -0.0294 +/- 0.0028   || 1882.93 / (1590-5),  p = 2.8e-07000 || 
 ||                      || BB-cf_z_0_10-0-broadband_sky-scale-sky || BB-cf_z_0_10-0-broadband_sky-sigma-sky || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias-hcd             ||                      || 
 ||                      ||                      ||                      ||                      ||                      ||                      ||                      ||                      ||                      || 
 || std+fitSky           || 0.000121 +/- 1.1e-050 || 11.51 +/- 0.62       || 24.74 +/- 0.60       || 1.494 +/- 0.064      || 0.84 +/- 0.17        || -0.206 +/- 0.0040    || -0.0259 +/- 0.0024   || 1624.98 / (1590-7),  p = 0.23 || 
 ||                      || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias-hcd             ||                      || 
 ||                      ||                      ||                      ||                      ||                      ||                      ||                      || 
 || std+rp>4             || 24.92 +/- 0.62       || 1.480 +/- 0.063      || 0.78 +/- 0.17        || -0.2071 +/- 0.0041   || -0.0249 +/- 0.0025   || 1599.01 / (1547-5),  p = 0.15 ||
```

lya_sky

lya_sky_2

**Comment by ngbusca** [comment]
Beautiful!! It's `r**2 \xi` right? what do the standard wedges look like with the new sky broadband?

**Comment by ngbusca** [comment]
@londumas please post the standard wedges with the new fit and merge when ready!

**Comment by ngbusca** [comment]
Ok, so beta, the sky broadband and the HCD model are completely degenerate... we'll have to think how to deal with that.

**Comment by londumas** [comment]
xi_mu_3

xi_mu_2

xi_mu_1

xi_mu_0




```
 ||                      || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias_NV(1239)        || bias_NV(1243)        || bias_SiII(1190)      || bias_SiII(1260)      || bias_SiIII(1207)     || bias-hcd             ||                      || 
 || std                  || 24.76 +/- 0.74       || 1.737 +/- 0.080      || 0.68 +/- 0.17        || -0.2283 +/- 0.0040   || -0.00272 +/- 0.00066 || -0.0004 +/- 0.0013   || -0.0013 +/- 0.0013   || -0.00074 +/- 0.00055 || -0.0012 +/- 0.00070  || -0.0294 +/- 0.0035   || 1822.18 / (1590-10),  p = 1.9e-050 || 

 ||                      || BB-cf_z_0_10-0-broadband_sky-scale-sky || BB-cf_z_0_10-0-broadband_sky-sigma-sky || L0-hcd               || \alpha_{\parallel}   || \alpha_{\perp}       || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias_NV(1239)        || bias_NV(1243)        || bias_SiII(1190)      || bias_SiII(1260)      || bias_SiIII(1207)     || bias-hcd             ||                      || 
 || std+fitSky           || 0.000120 +/- 1.1e-050 || 11.56 +/- 0.62       || 24.96 +/- 0.73       || 1.028 +/- 0.033      || 0.974 +/- 0.041      || 1.449 +/- 0.066      || 0.84 +/- 0.17        || -0.2046 +/- 0.0041   || -0.00282 +/- 0.00067 || -0.0006 +/- 0.0013   || -0.0013 +/- 0.0013   || -0.00098 +/- 0.00057 || -0.00108 +/- 0.00070 || -0.0261 +/- 0.0031   || 1193.54 / (1252-14),  p = 0.81 || 

 ||                      || L0-hcd               || \beta_{\mathrm{Ly}\alpha} || beta-hcd             || b_{\eta\,\mathrm{Ly\alpha}} = (b\beta)_{\mathrm{Ly\alpha}}/f || bias_NV(1239)        || bias_NV(1243)        || bias_SiII(1190)      || bias_SiII(1260)      || bias_SiIII(1207)     || bias-hcd             ||                      || 
 || std+rp>4             || 24.92 +/- 0.79       || 1.445 +/- 0.063      || 0.78 +/- 0.17        || -0.2064 +/- 0.0041   || -0.00278 +/- 0.00066 || -0.0006 +/- 0.0013   || -0.0012 +/- 0.0013   || -0.00084 +/- 0.00056 || -0.00143 +/- 0.00069 || -0.0245 +/- 0.0031   || 1534.80 / (1547-10),  p = 0.51 ||
```

**Comment by londumas** [comment]
@ngbusca, ok to merge?

**Comment by londumas** [comment]
@ngbusca, or we wait for DR14 papers?

**Comment by londumas** [comment]
As asked by @andreufont, here is the plot for Lya(Lya) x Lya(CIV) and for MgII(MgII)xMgII(MgII).

sky_substraction_a_la_DR12

mgII_auto

**Pull request merged**